### PR TITLE
Fix broken 64bit hash in tbb in 32bit builds

### LINKED ISSUE #181: testSymbolicFactorGraph unit test failure
As part of debugging #168, I found a new issue running the unit tests for i386 (for some reason it works right on amd64): 

```
 68/237 Test  #68: testSymbolicFactorGraph ................***Failed    0.08 sec
Not equal:
expected:
: cliques: 4, variables: 6
expected:
- P( e0 l0 b0)
expected:
| - P( s0 | b0 l0)
expected:
| - P( t0 | e0 l0)
expected:
| - P( x0 | e0)
actual:
: cliques: 4, variables: 6
actual:
- P( e0 l0 b0)
actual:
| - P( t0 | e0 l0)
actual:
| - P( x0 | e0)
actual:
| - P( s0 | b0 l0)
/home/jlblanco/code/gtsam/gtsam/symbolic/tests/testSymbolicFactorGraph.cpp:100: Failure: "assert_equal(asiaBayesTree, actual2)" 
There were 1 failures
```

@dellaert  Any idea where to start looking at to debug?

### PR DESCRIPTION
**Pull request by jlblancoc**
Fixes #181

---
This change is [Reviewable](https://reviewable.io/reviews/borglab/gtsam/186)

### CODE REVIEW

**Comment by jlblancoc** [comment]
do not merge yet, there's a stupid typo, and also the error showed that we don't have any configuration in Travis with tbb enabled, this could be a good moment to add it...

**Comment by jlblancoc** [comment]
This PR is finally passing Travis. Please, revise and merge at your convenience.

File: gtsam/base/ConcurrentMap.h [hunk 1]
```diff
@@ -29,14 +29,22 @@
 #  undef max
 #  undef ERROR
 
+#include <functional> // std::hash()
+
 // Use TBB concurrent_unordered_map for ConcurrentMap
-#  define CONCURRENT_MAP_BASE tbb::concurrent_unordered_map<KEY, VALUE>
+template <typename KEY, typename VALUE>
+using CONCURRENT_MAP_BASE = tbb::concurrent_unordered_map<
```
Comment (dellaert) [inline gtsam/base/ConcurrentMap.h]: Naming: I think we have to use ConcurrentMapBase now that it is no longer a macro. I don’t think any projects would use the Macro, so backwards compatibility is not an issue

File: gtsam/base/ConcurrentMap.h [hunk 2]
```diff
@@ -29,14 +29,22 @@
...
+  KEY,
+  VALUE,
+  std::hash<KEY>
```
Comment (dellaert) [inline gtsam/base/ConcurrentMap.h]: Nice!

File: gtsam/base/ConcurrentMap.h (hunk 1 above)
Comment (jlblancoc) [reply gtsam/base/ConcurrentMap.h]: Sure, I also like that name more.

**Comment by dellaert** [comment]
Yay !!!

**Pull request merged**